components/Common/Pagination/useGetPageElements.tsx

```diff
@@ -0,0 +1,146 @@
+'use client';
+import Link from 'next/link';
+import { useIntl } from 'react-intl';
+
+import styles from './index.module.css';
+
+import type { PaginationProps } from '.';
+
+type ParsedPage = PaginationProps['pages'][number] & {
+  pageNumber: number;
+};
+
+function parsePages(pages: PaginationProps['pages']): ParsedPage[] {
+  return pages.map(({ url }, index) => ({
+    url,
+    pageNumber: index + 1,
+  }));
+}
+
+const ellipsis = (
+  <span aria-hidden="true" className={styles.ellipsis}>
+    ...
+  </span>
+);
+
+function createPageElements({
+  parsedPages,
+  currentPage,
+  intl,
+}: Pick<PaginationProps, 'currentPage'> & {
+  parsedPages: ParsedPage[];
+  intl: ReturnType<typeof useIntl>;
+}) {
+  return parsedPages.map(({ url, pageNumber }) => {
+    return (
+      <li
+        key={pageNumber}
+        aria-setsize={parsedPages.length}
+        aria-posinset={pageNumber}
+      >
+        <Link
+          prefetch={false}
+          href={url}
+          aria-label={intl.formatMessage(
+            {
```

Comment:
These two objects probably can also be declared in one line, no?

Reply:
By this you mean that they can be left like this?

```ts
aria-label={intl.formatMessage({ id: 'components.common.pagination.pageLabel' }, { pageNumber })}
```

If so, yes, I can change it.

Reply:
No, sorry, I meant instead of:

```tsx
{
  pageNumber
}
```

it can be

```tsx
{ pageNumber }
```

(same for the id one)

Reply:
Ohh, okay, great. I will add it this change in an upcoming commit. Thank you!